cluster-autoscaler/cloudprovider/clusterapi/clusterapi_utils.go

```diff
@@ -153,3 +162,46 @@ func normalizedProviderString(s string) normalizedProviderID {
 	split := strings.Split(s, "/")
 	return normalizedProviderID(split[len(split)-1])
 }
+
+func scaleFromZeroEnabled(annotations map[string]string) bool {
+	cpu := annotations[cpuKey]
+	mem := annotations[memoryKey]
+
+	if cpu != "" && mem != "" {
+		return true
+	}
+	return false
+}
+
+func parseKey(annotations map[string]string, key string) (resource.Quantity, error) {
+	if val, exists := annotations[key]; exists && val != "" {
+		return resource.ParseQuantity(val)
+	}
+	return zeroQuantity.DeepCopy(), nil
+}
+
+func parseCPUCapacity(annotations map[string]string) (resource.Quantity, error) {
+	return parseKey(annotations, cpuKey)
+}
+
+func parseMemoryCapacity(annotations map[string]string) (resource.Quantity, error) {
+	// the value for the memoryKey is expected to have the unit type included,
+	// eg "1024Mi". if only a number is present, we add the suffix "Mi".
+	val, exists := annotations[memoryKey]
+	if exists && val != "" {
+		if _, err := strconv.Atoi(val); err == nil {
+			// value is a number, we will append "Mi" as the unit type
+			val = fmt.Sprintf("%sMi", val)
```

Comment:
I guess this is because the cloud provider controllers aren't labelling them correctly? To me, this feels like we should fix that in the labelling rather than here, WDYT @elmiko @enxebre ?

My consideration is we should try to keep the format the same as expected by Kubernetes itself (https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-memory) and shouldn't be expecting it to be a number of megabytes

Reply:
++ agreed about the provider using values that follow the kubernetes guide.
currently the aws provider is just using a number, i'll take a look at that code to see if we could improve it.

Reply:
Yes, I'd like to get this PR merged in a way which support compatibility i.e as it's now (checks if the value has a format and append otherwise).
Then merge the e2e PR.
Then do the changes on the actuators to set M/Mi driven by e2e running

Reply:
Cool, so long as there is a plan and we don't lose track, happy for this change to be a follow up

Reply:
i will add a TODO note here